manifests/init.pp

```diff
@@ -94,7 +94,9 @@
   if $control_enable {
     Service[$service_name] {
       restart   => 'unbound-control reload',
+      require   => Class['unbound::remote'],
     }
+    Package<| title == $package_name |> -> Class['unbound::remote']
```

Comment:
Nope, it there is no match then it will not create a dependency.  It will silently "fail".